module/filter/basic_kalman.cpp

```diff
@@ -2,4 +2,55 @@
 
 namespace basic_kalman {
 
+RM_kalmanfilter::RM_kalmanfilter() : KF_(4, 2) {
+  measurement_matrix   = cv::Mat::zeros(2, 1, CV_32F);
+  KF_.transitionMatrix = (cv::Mat_<float>(4, 4) << 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1);
+
+  setIdentity(KF_.measurementMatrix, cv::Scalar::all(1));
+  setIdentity(KF_.processNoiseCov, cv::Scalar::all(1e-3));
+  setIdentity(KF_.measurementNoiseCov, cv::Scalar::all(1e-2));  // 测量协方差矩阵R，数值越大回归越慢
+  setIdentity(KF_.errorCovPost, cv::Scalar::all(1));
+
+  KF_.statePost = (cv::Mat_<float>(4, 1) << x, y, 0, 0);  // 初始值
+}
+
+RM_kalmanfilter::~RM_kalmanfilter() {}
+
+cv::Point2f RM_kalmanfilter::predict_point(cv::Point2f _p) {
+  // p = _p;
+  cv::Mat     prediction = KF_.predict();
+  cv::Point2f predict_pt = cv::Point2f(prediction.at<float>(0), 0);
+
+  measurement_matrix.at<float>(0, 0) = _p.x;
+
+  KF_.correct(measurement_matrix);
+  return predict_pt;
+}
+
+void RM_kalmanfilter::reset() { measurement_matrix = cv::Mat::zeros(2, 1, CV_32F); }
+
+//
+float RM_kalmanfilter::use_RM_KF(float top) {
+  // top_angle_differ.top_angle_ 表示这一时刻的陀螺仪数值
+  // top_angle_differ.top_angle  表示上一时刻的陀螺仪数值
+  top_angle_differ.top_angle_ = top;
+
+  // 第一次获取的陀螺仪数据时, 对上一时刻(不存在)的陀螺仪数据的假设
+  if (top_angle_differ.get_top_times == 0) {
+    top_angle_differ.top_angle = top;
+  }
+
+  std::cout << "top: " << top << std::endl;
+
+  top_angle_differ.get_top_times++;  // 自动计数 . 获得陀螺仪数据的次数
```

Comment:
```suggestion
  ++top_angle_differ.get_top_times;  // 自动计数 . 获得陀螺仪数据的次数
```

为了保证风格统一，显式使用前置的 `++`。如果在表达式中，无非必要建议使用前置的 `++` 避免临时值的生成：

```asm
// num++
mov     DWORD PTR [rbp-20], edi
mov     eax, DWORD PTR [rbp-20]
lea     edx, [rax+1]
mov     DWORD PTR [rbp-20], edx
mov     DWORD PTR [rbp-4], eax
mov     eax, DWORD PTR [rbp-20]

// ++num
mov     DWORD PTR [rbp-20], edi
add     DWORD PTR [rbp-20], 1
mov     eax, DWORD PTR [rbp-20]
mov     DWORD PTR [rbp-4], eax
mov     eax, DWORD PTR [rbp-20]
```